### PR TITLE
DEV030 Create index and show pages for organization model

### PR DESCRIPTION
**Pull request by Maxxx911**
[card](https://trello.com/c/aQwNX4vT)
Create index and show pages for organization model

### CODE REVIEW

File: app/javascript/components/organizations/organization_card.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import CardActionArea from '@material-ui/core/CardActionArea';
+import CardActions from '@material-ui/core/CardActions';
+import CardContent from '@material-ui/core/CardContent';
+import CardMedia from '@material-ui/core/CardMedia';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+
+function make_url(props){
+  return 'organizations/' + props.organization.id
+
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_card.jsx]: delete new line

File: app/javascript/components/organizations/organization_card.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_card.jsx]: delete new line

File: app/javascript/components/organizations/organization_card.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        </CardActionArea>
+        <CardActions>
+          <Button size="small" color="primary">
+            Learn More
+          </Button>
+        </CardActions>
+      </Card>
+    </a>
+  );
+}
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_card.jsx]: add new line + setup your code editor to add new line every time

File: app/javascript/components/organizations/organization_detail.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import landingPageStyle from "../shared/style/landingPage";
+import ImageCarousel from "../shared/image_carousel";
+import TypoGraphy from '@material-ui/core/Typography'
+import Box from '@material-ui/core/Box'
+import EventCard from '../shared/card_event'
+import MyImage from "../shared/image";
+import OrganizationTitle from './organization_title'
+
+class OrganizationDetail extends React.Component {
+    constructor(props) {
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_detail.jsx]: Are you sure that we need constructor in here? `constructor` usage is out dated

File: app/javascript/components/organizations/organization_detail.jsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+                  )}
+                </Grid>
+              </Grid>
+            </div>
+          </div>
+        );
+    }
+}
+
+export default withStyles(landingPageStyle)(OrganizationDetail);
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_detail.jsx]: add new line

File: app/javascript/components/organizations/organization_index.jsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import classNames from "classnames";
+import withStyles from "@material-ui/core/styles/withStyles";
+import landingPageStyle from "../shared/style/landingPage";
+import OrganizationCard from './organization_card'
+import Grid from '@material-ui/core/Grid'
+
+class OrganizationDetail extends React.Component {
+    constructor(props) {
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_index.jsx]: The same comment about `constructor`

File: app/javascript/components/organizations/organization_index.jsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      super(props);
+      this.state = {
+        organizations: props.organizations
+    };}
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_index.jsx]: add new line

File: app/javascript/components/organizations/organization_index.jsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+                  </Grid>
+                )}
+              </Grid>
+            </div>
+          </div>
+        );
+    }
+}
+
+export default withStyles(landingPageStyle)(OrganizationDetail);
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_index.jsx]: new line

File: app/javascript/components/organizations/organization_title.jsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import Grid from '@material-ui/core/Grid';
+import TypoGraphy from '@material-ui/core/Typography'
+import Box from '@material-ui/core/Box'
+import MediumButton from '../shared/button'
+
+class OrganizationTitle extends React.Component {
+    constructor(props) {
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_title.jsx]: `constructor`

File: app/javascript/components/organizations/organization_title.jsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+      super(props);
+      this.state = {  organization_name: props.organization_name,
+                      current_user: props.current_user,
+                      users: props.users
+      };
+    }
+
+    render() {
+      if(JSON.stringify(this.state.users).includes(JSON.stringify(this.state.current_user))){
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_title.jsx]: this is check if not readable. Create method for JSON and use decomposition

File: app/javascript/components/organizations/organization_title.jsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+        return (
+          <Grid container direction="row" justify="center" alignItems="center">
+            <Grid item md={6} xs={12}>
+              <TypoGraphy>
+                <Box component='text' fontSize="3em" fontWeight="300">
+                  {this.state.organization_name}
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_title.jsx]: decomposition

File: app/javascript/components/organizations/organization_index.jsx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+      this.state = {
+        organizations: props.organizations
+    };}
+    render() {
+        const {classes} = this.props;
+        return (
+          <div className={classNames(classes.main, classes.mainRaised)}>
+            <div className={classes.container}>
+              <Grid container direction="row" justify="center" alignItems="stretch" spacing={3}>
+                { this.state.organizations.map((organization, i) =>
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_index.jsx]: decomposition

File: app/javascript/components/organizations/organization_title.jsx [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+          </Grid>
+        );
+        }
+        else{
+          return (
+            <Grid container direction="column" justify="center" alignItems="center">
+              <Grid item>
+                <TypoGraphy>
+                  <Box component='text' fontSize="3em" fontWeight="300">
+                    {this.state.organization_name}
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_title.jsx]: decomposition

File: app/javascript/components/organizations/organization_title.jsx [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+                  </Box>
+                </TypoGraphy>
+              </Grid>
+            </Grid>
+          );
+        }
+    }
+}
+
+export default OrganizationTitle;
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_title.jsx]: new line

File: app/views/organizations/index.html.erb [hunk 15]
```diff
@@ -0,0 +1,2 @@
+<%= javascript_pack_tag 'application' %>
+<%= react_component("organizations/organization_index", { organizations: @organizations }) %>
```
Comment (andreydro) [inline app/views/organizations/index.html.erb]: new line

File: app/javascript/components/organizations/organization_title.jsx [hunk 16]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import Grid from '@material-ui/core/Grid';
+import TypoGraphy from '@material-ui/core/Typography'
+import Box from '@material-ui/core/Box'
+import MediumButton from '../shared/button'
+
+class OrganizationTitle extends React.Component {
+  current_user_is_ovner(){
```
Comment (andreydro) [inline app/javascript/components/organizations/organization_title.jsx]: spelling mistake - owner

Comment (andreydro) [reply app/javascript/components/organizations/organization_title.jsx]: This is very strange function. You don't need to return true or false.

Comment (andreydro) [reply app/javascript/components/organizations/organization_title.jsx]: Can we send data in way that we don't need to use `JSON.stringify`?

Comment (andreydro) [reply app/javascript/components/organizations/organization_title.jsx]: and use decomposition `this.props`

File: app/controllers/organizations_controller.rb [hunk 17]
```diff
@@ -3,12 +3,18 @@ class OrganizationsController < ApplicationController
   before_action :set_organization, only: %i[show update destroy]
```
Comment (als6k) [inline app/controllers/organizations_controller.rb]: - show

File: app/javascript/components/shared/card_event.jsx [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+      <CardActions>
+        <a href={props.link} data-method="get">
+          <Button size="small" color="primary">
+            Learn More
+          </Button>
+        </a>
+      </CardActions>
+    </Card>
+  );
+}
```
Comment (andreydro) [inline app/javascript/components/shared/card_event.jsx]: new line

**Pull request merged**